usaspending_api/awards/v2/views/federal_accounts_count.py

```diff
@@ -0,0 +1,51 @@
+import logging
+
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.awards.models import FinancialAccountsByAwards, Award
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.validator.tinyshield import TinyShield
+from usaspending_api.common.validator.award import get_internal_or_generated_award_id_model
+
+
+logger = logging.getLogger("console")
+
+
+class FederalAccountCountRetrieveViewSet(APIView):
+    """
+    This route sends a request to the backend to retrieve the number of federal accounts associated with the requested award
+    """
+
+    endpoint_doc = "usaspending_api/api_contracts/contracts/v2/awards/federal_account_count.md"
+
+    def _parse_and_validate_request(self, provided_award_id: str) -> dict:
+        request_dict = {"award_id": provided_award_id}
+        models = get_internal_or_generated_award_id_model()
+        return TinyShield([models]).block(request_dict)
+
+    def _business_logic(self, request_data: dict) -> list:
+
+        award_id = request_data["award_id"]
+        award_id_column = "award_id" if type(award_id) is int else "award__generated_unique_award_id"
+        filter = {award_id_column: award_id}
+
+        try:
```

Comment:
One idea:
this could be refactored to use `id` returned from Award be used for the next queryset filter.

```python
try:
    award_id_column = "id" if type(award_id) is int else "generated_unique_award_id"
    award = Award.objects.get(**{award_id_column: award_id})
except Award.DoesNotExist:
    "...."

federal_account_queryset = FinancialAccountsByAwards.objects.filter(award_id=award.id).count()
```